chips/sam4l/src/usart.rs

```diff
@@ -492,15 +494,15 @@ impl USART {
         self.usart_tx_state.set(USARTStateTX::Idle);
     }
 
-    fn abort_rx(&self, usart: &USARTRegManager, error: hil::uart::Error) {
+    fn abort_rx(&self, usart: &USARTRegManager, rcode: ReturnCode, error: uart::Error) {
         if self.usart_rx_state.get() == USARTStateRX::DMA_Receiving {
             self.disable_rx_interrupts(usart);
             self.disable_rx(usart);
             self.usart_rx_state.set(USARTStateRX::Idle);
 
             // get buffer
             let mut length = 0;
-            let buffer = self.rx_dma.get().map_or(None, |rx_dma| {
+            let mut buffer = self.rx_dma.get().map_or(None, |rx_dma| {
```

Comment:
use `and_then` instead of `map_or` here
```suggestion
            let mut buffer = self.rx_dma.get().and_then(|rx_dma| {
```